clippy_utils/src/source.rs

```diff
@@ -306,6 +306,199 @@ pub fn snippet_with_context(
     )
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ExprPosition {
```

Comment:
Can we use `ExprPrecedence` from rustc?

Reply:
Doesn't distinguish between the left and right. Given `a - b` replacing `a` with `x + y` doesn't require parenthesis, replacing `b` does.

Reply:
Here is how [pretty print](https://github.com/rust-lang/rust/blob/b0535508474fb4353a3b83ef79df87817f028a71/compiler/rustc_ast_pretty/src/pprust/state.rs#L1839-L1847) does it. I guess you'd still have to pass `is_rhs: bool` in some form so 🤷.

Reply:
Seams like there is no ideal solution. I think this way is acceptable, are you also fine with it @camsteffen?

Reply:
Yeah this works.